### PR TITLE
Stream subject transform improvements

### PR DESCRIPTION
**Pull request by jnmoyne**
Improve checking that subject filters and destination transforms are valid subjects.
Improve error messages when bad filters/transforms are encountered.
Cover all cases of updating an existing stream's subject transform or sources

/cc @nats-io/core

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -428,11 +428,18 @@ func (a *Account) addStreamWithAssignment(config *StreamConfig, fsConfig *FileSt
 	if len(cfg.Sources) > 0 {
 		for i, ssi := range cfg.Sources {
 			ssi.setIndexName(i)
+			// check the filter, if any, is valid
+			if ssi.FilterSubject != _EMPTY_ {
```
Comment (derekcollison) [inline server/stream.go:432]: Can put on one line.

```go
if ssi.FilterSubject != _EMPTY_ && !IsValidSubject(ssi.FilterSubject)
```

File: server/stream.go [hunk 2]
```diff
@@ -1592,17 +1624,39 @@ func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool)
...
 
+	// Check for changes to subject transform
+	if ocfg.SubjectTransform == nil && cfg.SubjectTransform != nil {
+		tr, err := NewSubjectTransform(cfg.SubjectTransform.Source, cfg.SubjectTransform.Destination)
+		if err != nil {
+			mset.mu.Unlock()
+			return fmt.Errorf("stream configuration for subject transform from '%s' to '%s' %w", cfg.SubjectTransform.Source, cfg.SubjectTransform.Destination, err)
+		}
+		mset.itr = tr
+	} else if ocfg.SubjectTransform != nil && cfg.SubjectTransform != nil && (ocfg.SubjectTransform.Source != cfg.SubjectTransform.Source || ocfg.SubjectTransform.Destination != cfg.SubjectTransform.Destination) {
```
Comment (derekcollison) [inline server/stream.go:1649]: Make multi-line?

File: server/stream.go (hunk 1 above)
Comment (jnmoyne) [reply server/stream.go]: Done

File: server/stream.go (hunk 2 above)
Comment (jnmoyne) [reply server/stream.go]: Done

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**